### PR TITLE
support union and distinct in new plan.

### PR DESCRIPTION
**Pull request by hanfei19910905**
@shenli @coocood @zimulala  PTAL

### CODE REVIEW

File: executor/new_executor.go [hunk 1]
```diff
@@ -1006,3 +1006,65 @@ func (e *TrimExec) Next() (*Row, error) {
 	row.Data = row.Data[:e.len]
 	return row, nil
 }
+
+// NewUnionExec represents union executor.
+type NewUnionExec struct {
+	fields []*ast.ResultField
+	schema expression.Schema
+	Projs  []Executor
```
Comment (coocood) [inline executor/new_executor.go]: Rename `Projs' to 'Srcs' is better.

File: executor/new_builder.go [hunk 2]
```diff
@@ -228,3 +228,16 @@ func (b *executorBuilder) buildTrim(v *plan.Trim) Executor {
...
 }
+
+func (b *executorBuilder) buildNewUnion(v *plan.NewUnion) Executor {
+	e := &NewUnionExec{
+		schema: v.GetSchema(),
+		fields: v.Fields(),
+		Projs:  make([]Executor, len(v.Selects)),
+	}
+	for i, sel := range v.Selects {
+		selExec := b.build(sel)
```
Comment (shenli) [inline executor/new_builder.go]: Check b.err

File: executor/new_executor.go (hunk 1 above)
Comment (shenli) [inline executor/new_executor.go]: +1

File: executor/new_executor.go [hunk 3]
```diff
@@ -1006,3 +1006,65 @@ func (e *TrimExec) Next() (*Row, error) {
...
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if row == nil {
+			e.cursor++
+			continue
+		}
+		if e.cursor != 0 {
+			for i := range row.Data {
+				// The column value should be casted as the same type of the first select statement in corresponding position
```
Comment (shenli) [inline executor/new_executor.go]: if e.cursor == 0, do we need to do this?

File: executor/new_executor.go [hunk 4]
```diff
@@ -1006,3 +1006,65 @@ func (e *TrimExec) Next() (*Row, error) {
...
+}
+
+// Close implements Executor Close interface.
+func (e *NewUnionExec) Close() error {
+	var err error
+	e.cursor = 0
+	for _, sel := range e.Projs {
+		er := sel.Close()
+		if er != nil {
+			err = errors.Trace(er)
```
Comment (shenli) [inline executor/new_executor.go]: Why not return error directly?

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

File: executor/new_executor.go [hunk 5]
```diff
@@ -1006,3 +1006,65 @@ func (e *TrimExec) Next() (*Row, error) {
...
+}
+
+// Close implements Executor Close interface.
+func (e *NewUnionExec) Close() error {
+	var err error
+	e.cursor = 0
+	for _, sel := range e.Srcs {
+		er := sel.Close()
+		if er != nil {
+			err = errors.Trace(er)
```
Comment (zimulala) [inline executor/new_executor.go]: `=` -> `:=`, remove the line 1061.

File: executor/new_executor.go [hunk 6]
```diff
@@ -1006,3 +1006,65 @@ func (e *TrimExec) Next() (*Row, error) {
...
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if row == nil {
+			e.cursor++
+			continue
+		}
+		if e.cursor != 0 {
+			for i := range row.Data {
+				// The column value should be casted as the same type of the first select statement in corresponding position
```
Comment (zimulala) [inline executor/new_executor.go]: Add . at the end of a line

**Comment by zimulala** [comment]
Rest LGTM

**Pull request merged**